### DISCUSSION TITLE
Better performance for Capture Card with HDR + Tone mapping

### DISCUSSION THREAD

**Discussion by danbezerra** [mpv-player/mpv#14224] [Ideas]
Today I can use VideoProcessor + MadVR to capture HDR/LLDV signals and tone mapping it with MadVR. The solution works great, however, it's limited to an older version of MadVR for marketing reasons. 

I believe there is an opportunity here for MPV: Improve support for Capture cards (BlackMagic DeckLink Quad HDMI), including HDR/DV, offering on top of it all greatness of existing LibPlacebo tone mapping. 

Right now it doesn't really work well. I could only capture it with the no-latency mode and even then, I got a lot of dropped frames. Given that MPV is less resource-hungry when compared to MadVR, I don't see why this would not be possible with MPV as well.

Here is the link to VideoProcessor, if anyone is interested in investigating it: http://videoprocessor.org/ 

And here is the link to the AVSForum discussion: https://www.avsforum.com/threads/videoprocessor.3206050/page-339?post_id=63327826#post-63327826

With a few devices now fully supporting P7 FEL (via CoreElec), I would love to plug it with MPV via a capture card and enjoy a complete FEL + MPV tone mapping solution.

**Comment by kasper93** [comment]
> marketing reasons
> I believe there is an opportunity here for MPV

I doubt this is really opportunity for mpv project. That said patches welcome, without hardware it would be hard to optimize for said hardware.

**Comment by quietvoid** [comment]
> With a few devices now fully supporting P7 FEL (via CoreElec), I would love to plug it with MPV via a capture card and enjoy a complete FEL + MPV tone mapping solution.

You'd unironically be better off using CoreElec to tonemap with VS10.

**Reply by danbezerra** [reply]
Unfortunately, this is not the case when the tonemapping target is SDR.BT2020. VS10 (on the UGOOS at least) will either do DV->HDR or DV-> SDR BT.709.

**Comment by spedagadi** [comment]
Sharing what I did to get mpv tonemap on a linux device.

[CM3588](https://www.friendlyelec.com/index.php?route=product/product&product_id=294) has a hdmi input spec'd at 4K@60.
<img width="4032" height="3024" alt="IMG_0541" src="https://github.com/user-attachments/assets/99d1a3f3-9240-450a-9bca-dc2daf483fd6" />

Did a bit of work to get the board up and running with ubuntu, installed rockchip's ffmpeg, compiled libpacebo, mpv 0.41.0

working command:

./mpv-0.41.0/build/mpv --audio=no --no-config --no-cache --untimed --profile=gpu-hq --no-demuxer-thread --audio-buffer=0 --vd-lavc-threads=1 --demuxer-lavf-analyzeduration=0.5 --demuxer-lavf-probesize=42 --demuxer-lavf-o-add=rtbufsize=2G --vf=format=colorlevels=limited:primaries=bt.2020:gamma=pq:colormatrix=bt.2020-ncl:chroma-location=mpeg2/4/h264 --vo=gpu-next --fullscreen=yes --scale=spline36 --scale-blur=0.981251 --dscale=spline36 --linear-downscaling=no --cscale=spline36 --tscale=oversample --hdr-compute-peak=yes --hdr-contrast-recovery=0.5 --allow-delayed-peak-detect=no --target-peak=50 --target-trc=bt.1886 --tone-mapping=spline --tone-mapping-max-boost=1 --target-prim=bt.2020 --target-contrast=auto --gamut-mapping-mode=perceptual --temporal-dither=yes --dither-depth=auto av://v4l2:/dev/video0

I am not an mpv expert (the comamnd can be optimised perhaps) but i mpv compiled with right ffmeg (for the board) that manages to pull hdmi input signal from an av device, apply dynamic tonemapping with gpu-next and a few filters. I am using a 32G ram board - it is cute, a bunch of ssd slots too. DTM works without frame drops for 24hz input

Same works for decklink quard card in windows - need ffmpeg compiled with decklink and modify the command. There is a GOTCHA, HDR input is forced as I dont get the device tell me what it isi receiving - HDR/SDR/other. So you will notice a video filter in the command --vf=format=colorlevels=limited:primaries=bt.2020:gamma=pq:colormatrix=bt.2020-ncl:chroma-location=mpeg2/4/h264

I would like to writeup in a dedicated github page - let me know what I must produce as instructions & evidence so as to make it easy to follow and replicate.

**Reply by YukiNeeSan** [reply]
https://github.com/mpv-player/mpv/wiki

> I would like to writeup in a dedicated github page - let me know what I must produce as instructions & evidence so as to make it easy to follow and replicate.
